### PR TITLE
[Infrastructure] Add Repositories and Datastores

### PR DESCRIPTION
**Pull request by youssefsoli**
This PR adds the repositories and data stores we require with our use cases. I also added the spring dependency so the code can be run locally.

### CODE REVIEW

**Review by feiwun — APPROVED** [review]

**Review by modaserMoj — DISMISSED** [review]
The added files correctly set up all the needed repositories for the project using FireBase.

File: src/main/java/ca/utoronto/fitbook/adapter/persistence/firebase/ExerciseFirebaseRepository.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public Exercise getById(String id) {
+        ApiFuture<DocumentSnapshot> future = datastore.getCollection("exercises").document(id).get();
+        try {
+            DocumentSnapshot document = future.get();
+            if (document.exists()) {
+                String type = document.getString("type");
+                if (Objects.equals(type, Exercise.ExerciseType.REPETITIVE.toString()))
+                    return document.toObject(RepetitiveExercise.class);
+                else if (Objects.equals(type, Exercise.ExerciseType.TEMPORAL.toString()))
+                    return document.toObject(TemporalExercise.class);
```
Comment (jpablo2002) [inline src/main/java/ca/utoronto/fitbook/adapter/persistence/firebase/ExerciseFirebaseRepository.java:46]: I believe that this section of code in the ExerciseFirebaseRepository may be violating the Open/Closed SOLID principle, as any extension for a new kind of exercise would require the direct modification of the repository and adding new conditions.

**Review by jpablo2002 — CHANGES_REQUESTED** [review]
Most of it looks great to me! I have the following concern though:

File: src/main/java/ca/utoronto/fitbook/adapter/persistence/firebase/ExerciseFirebaseRepository.java (hunk 1 above)
Comment (youssefsoli) [reply src/main/java/ca/utoronto/fitbook/adapter/persistence/firebase/ExerciseFirebaseRepository.java:46]: Sounds good, will push a change.

**Review by jpablo2002 — DISMISSED** [review]
Looks good to me!

**Review by Ludosss — DISMISSED** [review]

**Review by Ludosss — APPROVED** [review]

**Review by modaserMoj — APPROVED** [review]

**Review by jpablo2002 — APPROVED** [review]
Lgtm

**Pull request merged**